Libs/MRML/Core/vtkMRMLTransformableNode.cxx

```diff
@@ -117,7 +117,18 @@ void vtkMRMLTransformableNode::SetAndObserveTransformNodeID(const char *transfor
     transformNodeID = 0;
     }
 
+  vtkMRMLTransformNode* previousNode = this->GetParentTransformNode();
   this->SetAndObserveNodeReferenceID(this->GetTransformNodeReferenceRole(), transformNodeID);
+  if (tnode)
```

Comment:
SetAndObserveTransformNodeID is just a convenience method for setting a parent transform. If parent transform s set using general node reference API (calling SetAndObserveNodeReferenceID directly) then you miss the notification. If you want to make sure you don't miss an added reference then override vtkMRMLNode::OnNodeReferenceAdded.

There is already GetReferencingNodes() method in vtkMRMLScene that can give you a list of all nodes that refer to a particular node. So, it is not very nice that you add a less generic mechanism to get the same information.

I would recommend to not add these two new events, but only add a single NodeReferenceModified event to the MRML scene, which would be invoked when any node references in the scene are modified. Then in your code you could observe this event and retrieve the current list of referencing nodes for the transform.